src/ghci/mod.rs

```diff
@@ -767,16 +768,20 @@ impl Ghci {
     /// This outputs how long the compilation took (since `compilation_start`), runs eval and test
     /// commands (if compilation succeeded), and writes the error log.
     #[instrument(skip_all, level = "trace")]
-    async fn finish_compilation(
+    async fn finish_compilation<const N: usize>(
         &mut self,
         compilation_start: Instant,
         log: &mut CompilationLog,
-        event: LifecycleEvent,
+        events: [LifecycleEvent; N],
     ) -> miette::Result<()> {
         // Allow hooks to consume the error log by updating it before running the hooks.
         self.write_error_log(log).await?;
 
-        self.run_hooks(event, log).await?;
+        for event in events {
+            self.run_hooks(event, log).await?;
+        }
+
+        let event = events[N - 1];
```

Comment:
Is there a way to enforce that `N` is greater than 0?

Alternatively, another way to guarantee the same thing is to split the `events` argument into two arguments: `events: [LifecycleEvent]` (for all but the last event) and `event: LifecycleEvent` (for the very last event).  This is essentially a poor person's encoding of a non-empty list.

I don't feel strongly about this, though.

Reply:
Not possible yet apparently :(